sciplot/PlotBase.hpp

```diff
@@ -0,0 +1,339 @@
+// sciplot - a modern C++ scientific plotting library powered by gnuplot
+// https://github.com/sciplot/sciplot
+//
+// Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+// Copyright (c) 2018-2021 Allan Leal
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#pragma once
+
+// C++ includes
+#include <sstream>
+#include <vector>
+
+// sciplot includes
+#include <sciplot/Constants.hpp>
+#include <sciplot/Default.hpp>
+#include <sciplot/Enums.hpp>
+#include <sciplot/Palettes.hpp>
+#include <sciplot/StringOrDouble.hpp>
+#include <sciplot/specs/AxisLabelSpecs.hpp>
+#include <sciplot/specs/BorderSpecs.hpp>
+#include <sciplot/specs/DrawSpecs.hpp>
+#include <sciplot/specs/FontSpecsOf.hpp>
+#include <sciplot/specs/FillStyleSpecs.hpp>
+#include <sciplot/specs/GridSpecs.hpp>
+#include <sciplot/specs/HistogramStyleSpecs.hpp>
+#include <sciplot/specs/LegendSpecs.hpp>
+#include <sciplot/specs/LineSpecsOf.hpp>
+#include <sciplot/specs/DrawSpecs.hpp>
+#include <sciplot/specs/TicsSpecs.hpp>
+#include <sciplot/specs/TicsSpecsMajor.hpp>
+#include <sciplot/specs/TicsSpecsMajor.hpp>
+#include <sciplot/specs/TicsSpecsMinor.hpp>
+#include <sciplot/Utils.hpp>
+
+namespace sciplot {
+
+/// The class used to create a plot containing graphical elements.
+class PlotBase
+{
+  public:
+    /// Construct a default Plot object
+    PlotBase();
```

Comment:
If this will be a base class, please add add a virtual destructor. [Read here why](https://softwareengineering.stackexchange.com/questions/384738/why-does-the-base-class-need-to-have-a-virtual-destructor-here-if-the-derived-cl).